spatialmath/box.go

```diff
@@ -352,3 +355,84 @@ func separatingAxisTest(positionDelta, plane r3.Vector, halfSizeA, halfSizeB [3]
 	}
 	return sum
 }
+
+// TODO: add function description
+func (b *box) ToPointCloud(options map[string]interface{}) ([]r3.Vector, error) {
```

Comment:
I think this can just become `resolution int` since it should be the same in all dimensions

Reply:
float64, not int. Probably in units of points per mm?

Reply:
yes, good catch Peter

Reply:
ok sure, but we need to keep `map[string]interface{}` because this allows for empty inputs, i.e. the user can use default spacing if they don't want to thing about it